tests/integration/pilot/mcs/discoverability/discoverability_test.go

```diff
@@ -0,0 +1,238 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discoverability
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"sync"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/cluster"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/common"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+const (
+	serviceA = "svc-a"
+	serviceB = "svc-b"
+)
+
+var (
+	i      istio.Instance
+	testNS string
+	echos  echo.Instances
+
+	retryTimeout = retry.Timeout(1 * time.Minute)
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.CustomSetup).
+		RequireMinVersion(17).
+		RequireMinClusters(2).
+		Setup(func(ctx resource.Context) error {
+			crd, err := ioutil.ReadFile("../../testdata/mcs-serviceexport-crd.yaml")
+			if err != nil {
+				return err
+			}
+			return ctx.Config().ApplyYAML("", string(crd))
+		}).
+		Setup(istio.Setup(&i, func(ctx resource.Context, cfg *istio.Config) {
+			cfg.ControlPlaneValues = `
+values:
+  pilot:
+    env:
+      ENABLE_MCS_SERVICE_DISCOVERY: "true"`
+		})).
+		Setup(func(ctx resource.Context) error {
+			// Create a new namespace in each cluster.
+			ns, err := namespace.New(ctx, namespace.Config{
+				Prefix: "mcs",
+				Inject: true,
+			})
+			if err != nil {
+				return err
+			}
+			testNS = ns.Name()
+
+			// Create echo instances in each cluster.
+			echos, err = echoboot.NewBuilder(ctx).
+				WithClusters(ctx.Clusters()...).
+				WithConfig(echo.Config{
+					Service:   serviceA,
+					Namespace: ns,
+					Ports:     common.EchoPorts,
+				}).
+				WithConfig(echo.Config{
+					Service:   serviceB,
+					Namespace: ns,
+					Ports:     common.EchoPorts,
+				}).Build()
+			return err
+		}).
+		Run()
+}
+
+func TestClusterLocal(t *testing.T) {
+	framework.NewTest(t).
+		Features("traffic.mcs.servicediscovery").
+		Run(func(ctx framework.TestContext) {
+			dest := echos.GetOrFail(ctx, echo.Service(serviceB))
+			for index, c := range ctx.Clusters() {
+				src := echos.GetOrFail(ctx, echo.Service(serviceA).And(echo.InCluster(c)))
+
+				ctx.NewSubTestf("from cluster %d", index).RunParallel(func(ctx framework.TestContext) {
```

Comment:
nit: use `cluster.StableName()` or let `echotest` do that for you.

Reply:
I think this no longer applies.

Reply:
Why not? You get the benefit of more descriptive cluster names in test names (i.e. `primary` `remote` `cross-network-primary`)

Reply:
Because I'm no longer manually spinning up subtests. I'm now using the `echotest` utility as you suggested. Or am I missing something?